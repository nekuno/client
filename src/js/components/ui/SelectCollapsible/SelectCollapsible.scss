@import '../../../../scss/_partials/colors';

.select-collapsible {
  cursor: pointer;
  font-weight: normal;
  padding-bottom: 8px;
  text-align: center;
  width: 100%;

  .title {
    margin-bottom: 5px;
  }
  .arrow {
    color: $purple-main;
    font-weight: bold;
  }
  .collapsible {
    background: $white-background;
    padding: 0 5%;
    text-align: left;
    position: absolute;
    width: 90%;

    .option {
      color: $gray-secondary-text;

      &.option-selected {
        cursor: default;
        color: $gray-main-text;
      }
    }
  }
}