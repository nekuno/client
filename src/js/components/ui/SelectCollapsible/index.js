import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './SelectCollapsible.scss';
import translate from "../../../i18n/Translate";
import AvailabilityEdit from "../../Availability/AvailabilityEdit/";

@translate('SelectCollapsible')
export default class SelectCollapsible extends Component {

    static propTypes = {
        options       : PropTypes.array.isRequired,
        selected      : PropTypes.string.isRequired,
        title         : PropTypes.string,
        onClickSelectCollapsible: PropTypes.func
    };

    constructor(props) {
        super(props);

        this.handleClickToggle = this.handleClickToggle.bind(this);
        this.handleClickItem = this.handleClickItem.bind(this);

        this.state = {
            open: false,
            selected: props.selected
        };
    }

    handleClickToggle() {
        const {open} = this.state;

        this.setState({open: !open})
    }

    handleClickItem(id) {
        this.setState({
            open: false,
            selected: id,
        });

        this.props.onClickSelectCollapsible(id);
    }

    search(nameKey, myArray){
        for (var i=0; i < myArray.length; i++) {
            if (myArray[i].id === nameKey) {
                return myArray[i];
            }
        }
    }

    render() {
        const {strings, options, title} = this.props;
        const {open, selected} = this.state;

        const orderStrings = {
            work        : strings.work,
            shows       : strings.shows,
            restaurants : strings.shows,
            plans       : strings.shows,
            sports      : strings.sports,
            hobbies     : strings.sports,
            games       : strings.sports,
        };

        return (
            <div className={styles.selectCollapsible} onClick={this.handleClickToggle}>
                <div className={styles.title + ' small'}>
                    {strings.orderBy} {this.search(selected, options).text}
                </div>
                {open ?
                    <div className={styles.arrow + ' icon icon-chevron-up'}/>
                    :
                    <div className={styles.arrow + ' icon icon-chevron-down'}/>
                }
                {open ?
                    <div className={styles.collapsible}>
                        {options.map((option, index) => {
                            return option.id === selected ?
                                <div key={index} className={styles.option + ' ' + styles.optionSelected}>
                                    {option.text}
                                </div>
                                :
                                <div key={index} className={styles.option} onClick={this.handleClickItem.bind(this, option.id)}>
                                    {option.text}
                                </div>
                        })}
                    </div>
                    : null}
            </div>
        );
    }
}

SelectCollapsible.defaultProps = {
    strings: {
        orderBy     : 'Order by',
    }
};