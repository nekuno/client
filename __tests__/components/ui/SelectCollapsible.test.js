import React from 'react';
import SelectCollapsible from '../../../src/js/components/ui/SelectCollapsible/SelectCollapsible.js';
import styles from '../../../src/js/components/ui/SelectCollapsible/SelectCollapsible.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test SelectCollapsible component', () => {
    const options = [
        {
            id: 'compatibility',
            text: 'Compatibility'
        },
        {
            id: 'similarity',
            text: 'Similarity'
        },
        {
            id: 'coincidences',
            text: 'Coincidences'
        }
    ];
    const mockFn = jest.fn();
    const selectCollapsible = shallow(<SelectCollapsible selected={'compatibility'} options={options} title={'Order'} onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(SelectCollapsible).toBeDefined();
    });
    it('should render correctly', () => {
        expect(selectCollapsible).toMatchSnapshot();
    });
});
