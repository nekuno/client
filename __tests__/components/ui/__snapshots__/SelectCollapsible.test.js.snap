// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SelectCollapsible component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    onClickHandler={[MockFunction]}
    options={
      Array [
        Object {
          "id": "compatibility",
          "text": "Compatibility",
        },
        Object {
          "id": "similarity",
          "text": "Similarity",
        },
        Object {
          "id": "coincidences",
          "text": "Coincidences",
        },
      ]
    }
    selected="compatibility"
    strings={
      Object {
        "games": "Leisure",
        "hobbies": "Leisure",
        "plans": "Experience",
        "restaurants": "Experience",
        "shows": "Experience",
        "sports": "Leisure",
        "work": "Work",
      }
    }
    title="Order"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "locale": undefined,
      "onClickHandler": [MockFunction],
      "options": Array [
        Object {
          "id": "compatibility",
          "text": "Compatibility",
        },
        Object {
          "id": "similarity",
          "text": "Similarity",
        },
        Object {
          "id": "coincidences",
          "text": "Coincidences",
        },
      ],
      "selected": "compatibility",
      "strings": Object {
        "games": "Leisure",
        "hobbies": "Leisure",
        "plans": "Experience",
        "restaurants": "Experience",
        "shows": "Experience",
        "sports": "Leisure",
        "work": "Work",
      },
      "title": "Order",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "locale": undefined,
        "onClickHandler": [MockFunction],
        "options": Array [
          Object {
            "id": "compatibility",
            "text": "Compatibility",
          },
          Object {
            "id": "similarity",
            "text": "Similarity",
          },
          Object {
            "id": "coincidences",
            "text": "Coincidences",
          },
        ],
        "selected": "compatibility",
        "strings": Object {
          "games": "Leisure",
          "hobbies": "Leisure",
          "plans": "Experience",
          "restaurants": "Experience",
          "shows": "Experience",
          "sports": "Leisure",
          "work": "Work",
        },
        "title": "Order",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
